include/boost/multiprecision/detail/standalone_config.hpp

```diff
@@ -0,0 +1,71 @@
+///////////////////////////////////////////////////////////////
+//  Copyright 2010 - 2021 Douglas Gregor
+//  Copyright 2021 Matt Borland.
+//  Distributed under the Boost Software License, Version 1.0.
+//  See accompanying file LICENSE_1_0.txt or copy at https://www.boost.org/LICENSE_1_0.txt
+//
+//  Used to support configuration options depending on standalone context
+//  by providing either required support or disabling functionality  
+
+#ifndef BOOST_MP_STANDALONE_CONFIG_HPP
+#define BOOST_MP_STANDALONE_CONFIG_HPP
+
+// If any of the most frequently used boost headers are missing assume that standalone mode is supposed to be used
+#ifdef __has_include
+#if !__has_include(<boost/config.hpp>) || !__has_include(<boost/assert.hpp>) || !__has_include(<boost/lexical_cast.hpp>) || \
+    !__has_include(<boost/throw_exception.hpp>) || !__has_include(<boost/predef/other/endian.h>)
+#   ifndef BOOST_MP_STANDALONE
+#       define BOOST_MP_STANDALONE
+#   endif
+#endif
+#endif
+
+#ifndef BOOST_MP_STANDALONE
+
+#include <boost/config.hpp>
+#include <boost/integer.hpp>
+#include <boost/integer_traits.hpp>
+
+#else // Standalone mode
+
+// Prevent Macro sub
+#ifndef BOOST_PREVENT_MACRO_SUBSTITUTION
+#  define BOOST_PREVENT_MACRO_SUBSTITUTION
+#endif
+
+// int/float 128 types
+
+#ifdef (__SIZEOF_INT128__)
```

Comment:
This will more or less work for vanilla GCC, but if you look at Boost.Config the ways in which this gets defined (or undefined) are pretty complex.  Not sure if we need to worry too much - maybe the answer is "use Boost" if the configuration is wrong... only SNAFU is that everything may compile but be inefficient if __int128 wasn't enabled.